<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="This package generates expression specificty scores for a range of cell types in a ctd object from the EWCE package. If required, it can generate input files for MAGMA and SLDSR gene set enrichment analyses.">
<title>Top decile expression specificity • tdespec</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Top decile expression specificity">
<meta property="og:description" content="This package generates expression specificty scores for a range of cell types in a ctd object from the EWCE package. If required, it can generate input files for MAGMA and SLDSR gene set enrichment analyses.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">tdespec</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="tdespec">tdespec<a class="anchor" aria-label="anchor" href="#tdespec"></a>
</h1></div>
<!-- badges: start -->
<!-- badges: end -->
<p>The goal of tdespec is to generate cell specificity scores for scRNAseq dataset as described in the <a href="https://www.nature.com/articles/s41588-018-0129-5" class="external-link">Skene et al. (2018)</a>. This utilises the functionality of the Bioconductor package <a href="https://github.com/NathanSkene/EWCE/" class="external-link">EWCE</a> to generate a CTD object containing the specificity scores for each cell type in a snRNAseq dataset and generates files for gene set enrichment analyses packages <a href="https://ctg.cncr.nl/software/magma" class="external-link">MAGMA</a> and <a href="https://github.com/bulik/ldsc" class="external-link">LD Score regression</a>.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the development version of tdespec like so:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span><span class="op">)</span> <span class="op">{</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'devtools'</span><span class="op">)</span> <span class="op">}</span> </span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span></span>
<span><span class="fu">install_github</span><span class="op">(</span><span class="st">"Dazcam/tdespec"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="basic-workflow">Basic workflow<a class="anchor" aria-label="anchor" href="#basic-workflow"></a>
</h2>
<p>The basic <code>tdespec</code> workflow is a 4-stage process:</p>
<ol style="list-style-type: decimal">
<li>Create a vector of genes overlapping the MHC region of the human genome.</li>
<li>Obtain a a tibble of a gene reference file with MHC overlapping genes removed.</li>
<li>Create a ctd object from and snRNAseq gene expression matrix.</li>
<li>Create input files for gene enrichment tests (MAGMA and LDSR) and save to file.</li>
</ol>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dazcam.github.io/tdespec/">tdespec</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"pbmc_small"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in data("pbmc_small"): data set 'pbmc_small' not found</span></span>
<span><span class="va">mhc_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_mhc_genes.html">get_mhc_genes</a></span><span class="op">(</span>build <span class="op">=</span> <span class="st">'hg19'</span>, method <span class="op">=</span> <span class="st">'rest_api'</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Connecting to Ensembl server via REST API...</span></span>
<span><span class="co">#&gt; MHC genes detected: 382</span></span>
<span><span class="co">#&gt; MHC genes with unique HGNC ID detected: 374</span></span>
<span><span class="va">gene_coord</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_gene_coordinates.html">get_gene_coordinates</a></span><span class="op">(</span><span class="va">mhc_genes</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">external_name</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Dowloading MAGMA hg19 reference file ...</span></span>
<span><span class="va">ctd_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/create_ctd.html">create_ctd</a></span><span class="op">(</span><span class="va">seurat_small</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">data</span>, </span>
<span>                       meta_lev_1 <span class="op">=</span> <span class="va">seurat_small</span><span class="op">$</span><span class="va">cluster_level_1</span>,</span>
<span>                       meta_lev_2 <span class="op">=</span> <span class="va">seurat_small</span><span class="op">$</span><span class="va">cluster_level_2</span>,</span>
<span>                       outdir <span class="op">=</span> <span class="st">'test/'</span>,</span>
<span>                       mhc_genes <span class="op">=</span> <span class="va">mhc_genes</span>,</span>
<span>                       group_name <span class="op">=</span> <span class="st">'brain_study'</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Creating CTD object ...</span></span>
<span><span class="co">#&gt; MHC genes removed: 0</span></span>
<span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Running CPM ...</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Dropping uninformative genes cpm norm ...</span></span>
<span><span class="co">#&gt; Check 21310Check 1452</span></span>
<span><span class="co">#&gt; 1 core(s) assigned as workers (7 reserved).</span></span>
<span><span class="co">#&gt; Converting to sparse matrix.</span></span>
<span><span class="co">#&gt; Checking for non-expressed genes.</span></span>
<span><span class="co">#&gt; -2918 / 18392 non-expressed genes dropped</span></span>
<span><span class="co">#&gt; Checking for cells with no expressed genes.</span></span>
<span><span class="co">#&gt; DGE:: Limma...</span></span>
<span><span class="co">#&gt; 8,495 / 18,392 genes dropped @ DGE adj_pval_thresh &lt; 1e-05</span></span>
<span><span class="co">#&gt; Time difference of 4.611608 secs</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Gene counts:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; RAW:21310</span></span>
<span><span class="co">#&gt; RAW_NO_MHC:21310</span></span>
<span><span class="co">#&gt; CPM_DROP_GENES:9897</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Creating CTD object ...</span></span>
<span><span class="co">#&gt; 1 core(s) assigned as workers (7 reserved).</span></span>
<span><span class="co">#&gt; Converting to sparse matrix.</span></span>
<span><span class="co">#&gt; + Calculating normalized mean expression.</span></span>
<span><span class="co">#&gt; Converting to sparse matrix.</span></span>
<span><span class="co">#&gt; Converting to sparse matrix.</span></span>
<span><span class="co">#&gt; + Calculating normalized specificity.</span></span>
<span><span class="co">#&gt; Converting to sparse matrix.</span></span>
<span><span class="co">#&gt; Converting to sparse matrix.</span></span>
<span><span class="co">#&gt; Converting to sparse matrix.</span></span>
<span><span class="co">#&gt; Converting to sparse matrix.</span></span>
<span><span class="co">#&gt; Loading required namespace: ggdendro</span></span>
<span><span class="co">#&gt; + Saving results ==&gt;  test//ctd_brain_study.rda</span></span>
<span><span class="fu"><a href="reference/create_enrich_test_files.html">create_enrich_test_files</a></span><span class="op">(</span>ctd_path <span class="op">=</span> <span class="va">ctd_path</span>, </span>
<span>                         study_id <span class="op">=</span> <span class="st">'brain_study'</span>,</span>
<span>                         gene_coordinates <span class="op">=</span> <span class="va">gene_coord</span>,</span>
<span>                         outdir <span class="op">=</span> <span class="st">'test/'</span><span class="op">)</span></span>
<span><span class="co">#&gt; Creating MAGMA input files for cluster level 1 ...</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(hgnc)`</span></span>
<span><span class="co">#&gt; Creating SLDSR input files for cluster level: 1...</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(hgnc)`</span></span>
<span><span class="co">#&gt; Output files have been saved to: test/.</span></span>
<span><span class="co">#&gt; Done.</span></span>
<span><span class="co">#&gt; Creating MAGMA input files for cluster level 2 ...</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(hgnc)`</span></span>
<span><span class="co">#&gt; Creating SLDSR input files for cluster level: 2...</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(hgnc)`</span></span>
<span><span class="co">#&gt; Output files have been saved to: test/.</span></span>
<span><span class="co">#&gt; Done.</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing tdespec</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Darren Cameron <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/YOUR-ORCID-ID" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Darren Cameron.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
