---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# tdespec

<!-- badges: start -->
<!-- badges: end -->

The goal of tdespec is to generate cell specificity scores for scRNAseq dataset as 
described in the [Skene et al. (2018)](https://www.nature.com/articles/s41588-018-0129-5).
This utilises the functionality of the Bioconductor package 
[EWCE](https://github.com/NathanSkene/EWCE/) to generate a CTD object containing the specificity 
scores for each cell type in a snRNAseq dataset and generates files for gene set 
enrichment analyses packages [MAGMA](https://ctg.cncr.nl/software/magma) and 
[LD Score regression](https://github.com/bulik/ldsc). 

## Installation

You can install the development version of tdespec like so:

``` r
if (!require(devtools)) { install.packages('devtools') } 
library(devtools)
install_github("Dazcam/tdespec")
```

## Basic workflow

The basic `tdespec` workflow is a 4-stage process:

1. Create a vector of genes overlapping the MHC region of the human genome.
2. Obtain a a tibble of a gene reference file with MHC overlapping genes removed.
3. Create a ctd object from and snRNAseq gene expression matrix.
4. Create input files for gene enrichment tests (MAGMA and LDSR) and save to file.

```{r example}
library(tdespec)
library(dplyr)
data("pbmc_small")
mhc_genes <- get_mhc_genes(build = 'hg19', method = 'rest_api')
gene_coord <- get_gene_coordinates(mhc_genes %>% pull(external_name))
ctd_path <- create_ctd(seurat_small@assays$RNA@data, 
                       meta_lev_1 = seurat_small$cluster_level_1,
                       meta_lev_2 = seurat_small$cluster_level_2,
                       outdir = 'test/',
                       mhc_genes = mhc_genes,
                       group_name = 'brain_study')
create_enrich_test_files(ctd_path = ctd_path, 
                         study_id = 'brain_study',
                         gene_coordinates = gene_coord,
                         outdir = 'test/')
```

