<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Create gene expression specificity scores for scRNAseq data"><title>Create gene expression specificity scores for scRNAseq data — create_ctd • tdespec</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Create gene expression specificity scores for scRNAseq data — create_ctd"><meta property="og:description" content="Create gene expression specificity scores for scRNAseq data"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">tdespec</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"></ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Create gene expression specificity scores for scRNAseq data</h1>
      
      <div class="d-none name"><code>create_ctd.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Create gene expression specificity scores for scRNAseq data</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">create_ctd</span><span class="op">(</span></span>
<span>  matrix <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  meta_lev_1 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  meta_lev_2 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  outdir <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  mhc_genes <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  group_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  threads <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>matrix</dt>
<dd><p>Raw count gene matrix - needs to be genes x cell and annotation data</p></dd>


<dt>meta_lev_1</dt>
<dd><p>A vector of cell IDs for level 1 cell types</p></dd>


<dt>meta_lev_2</dt>
<dd><p>A vector of cell IDs for level 1 cell types</p></dd>


<dt>outdir</dt>
<dd><p>The output directory to store the ctd object</p></dd>


<dt>mhc_genes</dt>
<dd><p>A set of MHC genes to be removed</p></dd>


<dt>group_name</dt>
<dd><p>An string identifier for the dataset</p></dd>


<dt>threads</dt>
<dd><p>Number of cores to assign to process. Default is NULL,
which will auto detect total cores and estimate number to use.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A CTD object containing</p>


<p>directory where ctd object was stored</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">'seurat_small'</span>, <span class="st">'mhc_genes'</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">create_ctd</span><span class="op">(</span><span class="va">seurat_small</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">data</span>,</span></span>
<span class="r-in"><span>                     meta_lev_1 <span class="op">=</span> <span class="va">seurat_small</span><span class="op">$</span><span class="va">cluster_level_1</span>,</span></span>
<span class="r-in"><span>                     meta_lev_2 <span class="op">=</span> <span class="va">seurat_small</span><span class="op">$</span><span class="va">cluster_level_2</span>,</span></span>
<span class="r-in"><span>                     outdir <span class="op">=</span> <span class="st">'test'</span>,</span></span>
<span class="r-in"><span>                     mhc_genes <span class="op">=</span> <span class="va">mhc_genes</span>,</span></span>
<span class="r-in"><span>                     group_name <span class="op">=</span> <span class="st">'brain_study'</span>,</span></span>
<span class="r-in"><span>                     threads <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating CTD object ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> MHC genes removed: 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: SeuratObject</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: sp</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The legacy packages maptools, rgdal, and rgeos, underpinning the sp package,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> which was just loaded, will retire in October 2023.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Please refer to R-spatial evolution reports for details, especially</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> https://r-spatial.org/r/2023/05/15/evolution4.html.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> It may be desirable to make the sf package available;</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> package maintainers should consider adding sf to Suggests:.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The sp package is now running under evolution status 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>      (status 2 uses the sf package in place of rgdal)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running CPM ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping uninformative genes cpm norm ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Check 21310Check 1452</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Warning: sctSpecies_origin not provided. Setting to 'mouse' by default.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 core(s) assigned as workers (1 reserved).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting to sparse matrix.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for non-expressed genes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -2918 / 18392 non-expressed genes dropped</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for cells with no expressed genes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> DGE:: Limma...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 8,495 / 18,392 genes dropped @ DGE adj_pval_thresh &lt; 1e-05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 3.894935 secs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene counts:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> RAW:21310</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> RAW_NO_MHC:21310</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> CPM_DROP_GENES:9897</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating CTD object ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 core(s) assigned as workers (1 reserved).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting to sparse matrix.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Calculating normalized mean expression.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting to sparse matrix.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting to sparse matrix.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Calculating normalized specificity.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting to sparse matrix.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting to sparse matrix.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting to sparse matrix.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting to sparse matrix.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required namespace: ggdendro</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Saving results ==&gt;  test/ctd_brain_study.rda</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "test/ctd_brain_study.rda"</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Darren Cameron.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

